File: micronaut-mcp-server-java-sdk/src/test/java/io/micronaut/mcp/server/stateless/sync/tools/HelloWorldTool.java
package io.micronaut.mcp.server.stateless.sync.tools;


import io.micronaut.context.annotation.Requires;
import io.micronaut.mcp.annotations.Tool;
import jakarta.inject.Singleton;

@Requires(property = "spec.name", value = "ToolAnnotationsTest")
//tag::clazz[]
@Singleton
class HelloWorldTool {
    @Tool(title = "Hello World",
        annotations = @Tool.ToolAnnotations(readOnlyHint = true,
            title = "Hello World",
            destructiveHint = false,
            idempotentHint = true,
            openWorldHint = false,
            returnDirect = true))
    String helloWorld() {
        return "Hello, World!";
    }
}
//end::clazz[]

File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/tools/HelloWorldTool.py
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.mcp.annotations import Tool


@Requires(property="spec.name", value="HelloWorldToolTest")
# tag::clazz[]
@Singleton
class HelloWorldTool:
    @Tool(title="Hello World",
          annotations=Tool.ToolAnnotations(readOnlyHint=True,
                                           title="Hello World",
                                           destructiveHint=False,
                                           idempotentHint=True,
                                           openWorldHint=False,
                                           returnDirect=True))
    def hello_world(self) -> str:
        return "Hello, World!"
# end::clazz[]
